accord-core/src/main/java/accord/local/NodeTimeService.java

```diff
@@ -36,18 +37,53 @@ public interface NodeTimeService
     long now();
 
     /**
-     * Return the current time since the Unix epoch in the specified time unit. May still be simulated time and not
-     * real time.
+     * Return the current time since some arbitrary epoch in the specified time unit. May still be simulated time and not
+     * real time. The time returned by this will be monotonic.
      */
-    long unix(TimeUnit unit);
+    long elapsed(TimeUnit unit);
 
     Timestamp uniqueNow(Timestamp atLeast);
 
-    static ToLongFunction<TimeUnit> unixWrapper(TimeUnit sourceUnit, LongSupplier nowSupplier)
+    static ToLongFunction<TimeUnit> elapsedWrapperFromMonotonicSource(TimeUnit sourceUnit, LongSupplier monotonicNowSupplier)
+    {
+        return resultUnit ->  resultUnit.convert(monotonicNowSupplier.getAsLong(), sourceUnit);
+    }
+
+    /**
+     * Allow time progression to be controlled using a potentially non-monotonic time source for testing with simulated
+     * time sources. This is not designed to be fast and real usage should be with a monotonic time source.
+     */
+    @VisibleForTesting
+    static ToLongFunction<TimeUnit> elapsedWrapperFromNonMonotonicSource(TimeUnit sourceUnit, LongSupplier nonMonotonicNowSupplier)
```

Comment:
is this really needed?  In the callers found they are actually monotonic... I assume C* is going to use `nanoTime`, so we don't have a `non-monotonic` case; and if we do thats actually a bug in the test code...

Reply:
If we are concerned about non-monotonic time won't that be something that test code such as `BurnTest` exercises? I wanted the passage of monotonic time to match non-monotonic time, but just be monotonic and based off of a different epoch similar to how `nanoTime` works.

We have code in `Node.uniqueNow` to ensure that it remains monotonic even when the time that is provided so we should exercise it if we aren't already.